test/functional/provider/python_spec.lua

```diff
@@ -1,19 +1,19 @@
+local helpers = require('test.functional.helpers')
+local eval, command, feed = helpers.eval, helpers.command, helpers.feed
+local eq, clear, insert = helpers.eq, helpers.clear, helpers.insert
+local expect, write_file = helpers.expect, helpers.write_file
+
 do
-  local proc = io.popen(
-    [[python -c 'import neovim, sys; sys.stdout.write("ok")' 2> /dev/null]])
-  if proc:read() ~= 'ok' then
+  command('let [interp, errors] = provider#pythonx#Detect(2)')
```

Comment:
This is setting globals `g:interp`, `g:errors`. Would be better to be explicit about that. The names also seem a bit risky for globals, something like `g:pydetect_errors` would be less risky.